File: chapter01/chapter01/node_tuto/src/minimal_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/Float64.h>

int main(int argc, char **argv){
	ros::init(argc,argv, "minimal_publisher");
	ros::NodeHandle n;

	ros::Publisher my_publisher_object = n.advertise<std_msgs::Float64>("topic1",1);

	std_msgs::Float64 input_float;

	input_float.data = 0.0;

	while(ros::ok())
	{
		input_float.data = input_float.data + 0.001;
		my_publisher_object.publish(input_float);
	}
}

File: Code/chapter01/node_tuto/scripts/minimal_publisher.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Float64

rospy.init_node("min_pub")
pub = rospy.Publisher("topic1", Float64, queue_size=1)

rate = rospy.Rate(1)
msg = Float64()
msg.data = 0.0

while not rospy.is_shutdown():
	pub.publish(msg)
	msg.data = msg.data + 0.1
	rate.sleep()

	